text_generator.go

```diff
@@ -17,22 +17,33 @@ type textPlistGenerator struct {
 	depth  int
 
 	dictKvDelimiter, dictEntryDelimiter, arrayDelimiter []byte
+
+	meta *Meta
 }
 
 var (
 	textPlistTimeLayout = "2006-01-02 15:04:05 -0700"
 	padding             = "0000"
 )
 
-/*
-	generateDocument(pVal cfValue, meta Meta)
- */
-
 func (p *textPlistGenerator) generateDocument(pval cfValue) {
-	p.writePlistValue(pval)
+	//if p.meta != nil && len(p.meta.Nodes) > 0 {
+	//	nodes = p.meta.Nodes[0].Nodes() // First is always the root of the document.
+	//}
+	if p.meta == nil {
+		p.writePlistValue(pval, nil)
```

Comment:
Is this the same as calling `p.writePlistValue(pval)`? If so I think it's cleaner to not pass in `nil`